[IMP] inventory: refactor WH and locations (#10238)
odoo/documentation: content/applications/inventory_and_mrp/inventory/warehouses_storage/inventory_management/warehouses.rst

==========
Warehouses
==========

In the Odoo *Inventory* app, a *warehouse* is a physical space with an address for storing items,
such as a storage facility, distribution center, or physical store.

Each database has a pre-configured warehouse with the company's address. Users can set up multiple

> warehouses and :doc:`create stock moves <use_routes>` between them.

Reviewer: ```suggestion
warehouses, and :doc:`create stock moves <use_routes>` between them.
```

Configuration
=============

To create or manage warehouses, go to :menuselection:`Inventory app --> Configuration -->
Warehouses`.

> Then, select an existing warehouse or create a new one by clicking :guilabel:`New`. Doing so opens

Reviewer: ```suggestion
Then, select an existing warehouse, or create a new one by clicking :guilabel:`New`. Doing so opens
```

the warehouse form, which contains the following fields:

> - :guilabel:`Warehouse`: (*required field*) the full name of the warehouse.
> - :guilabel:`Short Name`: (*required field*) the abbreviated code for the warehouse (maximum five
>   characters). The short name for the default warehouse in Odoo is `WH`.
> 
>   .. important::
>      The :guilabel:`Short Name` appears on warehouse documents, so it is recommended to use an
>      understandable one like "WH[first letters of location]", e.g. `WHA`, `WHB`.
> 
> - :guilabel:`Company`: (*required field*) the company that owns the warehouse; this can be set as
>   the company that owns the Odoo database or the company of a customer or vendor.
> - :guilabel:`Address`: (*required field*) the address of the warehouse. To change the warehouse
>   address when creating two or more warehouses, hover over the field and click the
>   :icon:`fa-arrow-right` :guilabel:`(right arrow)`.

Reviewer: ```suggestion
- :guilabel:`Warehouse` (*required field*):  the full name of the warehouse.
- :guilabel:`Short Name` (*required field*): the abbreviated code for the warehouse (maximum five
  characters). The short name for the default warehouse in Odoo is `WH`.

  .. important::
     The :guilabel:`Short Name` appears on warehouse documents, so it is recommended to use an
     memorable one, like "WH[first letters of location]" (e.g. `WHA`, `WHB`, etc.).

- :guilabel:`Company` (*required field*): the company that owns the warehouse; this can be set as
  the company that owns the Odoo database, or the company of a customer or vendor.
- :guilabel:`Address` (*required field*): the address of the warehouse. To change the warehouse
  address when creating two or more warehouses, hover over the field, and click the
  :icon:`fa-arrow-right` :guilabel:`(right arrow)`.
```
suggesting putting the 'required field' parenthetical **before** the colon/explanation, as it appears a bit cleaner this way -- totally your call, though

Reviewer: I'm also seeing an "Intrastat region" field here in the runbot I'm using for the review. So, I'm not sure if you want to also mention that (or mention why/how that would/wouldn't appear)... just something to consider during your revisions. 👍 
![Screenshot 2024-07-18 131821](https://github.com/user-attachments/assets/a14a1507-44e8-4f83-ba43-6602d6233973)

.. important::
   The options below are available **only** when the *Multi-Step Routes* feature is enabled in
   :menuselection:`Inventory app --> Configuration --> Settings`.

- :guilabel:`Incoming Shipments`: select the option to receive products from the warehouse in
  :doc:`one <../../shipping_receiving/daily_operations/receipts_delivery_one_step>`, :doc:`two
  <../../shipping_receiving/daily_operations/receipts_delivery_two_steps>`, or :doc:`three
  <../../shipping_receiving/daily_operations/receipts_three_steps>` steps.

- :guilabel:`Outgoing Shipments`: select the option to deliver products from the warehouse in
  :doc:`one <../../shipping_receiving/daily_operations/receipts_delivery_one_step>`, :doc:`two
  <../../shipping_receiving/daily_operations/receipts_delivery_two_steps>`, or :doc:`three
  <../../shipping_receiving/daily_operations/delivery_three_steps>` steps.

- :guilabel:`Dropship Subcontractors`: available with the *Subcontracting* feature enabled in
  :menuselection:`Manufacturing app --> Configuration --> Settings`. Tick this checkbox to purchase

>   components from vendors and dropship them to subcontractors.

Reviewer: ```suggestion
  components from vendors, and dropship them to subcontractors.
```

- :guilabel:`Resupply Subcontractors`: available with the *Subcontracting* feature, tick this

>   checkbox to supply subcontractors with raw materials stored in **this** warehouse.

Reviewer: ```suggestion
  checkbox to supply subcontractors with raw materials stored in *this* specific warehouse.
```

> - :guilabel:`Manufacture to Resupply`: tick this box to allow for items to be manufactured in this

Reviewer: ```suggestion
- :guilabel:`Manufacture to Resupply`: tick this checkbox to allow for items to be manufactured in this
```

  warehouse.
- :guilabel:`Manufacture`: choose whether to manufacture products in :doc:`one
  <../../../manufacturing/management/one_step_manufacturing>`, :doc:`two
  <../../../manufacturing/management/two_step_manufacturing>`, or :doc:`three steps
  <../../../manufacturing/management/three_step_manufacturing>`.

> - :guilabel:`Buy to Resupply`: check the box to allow for purchased products to be delivered to the

Reviewer: ```suggestion
- :guilabel:`Buy to Resupply`: tick this checkbox to allow for purchased products to be delivered to the
```

  warehouse.
- :guilabel:`Resupply From`: available with multiple warehouses in the database, select warehouses

>   to pull stock **from** to fulfill orders.

Reviewer: ```suggestion
  to pull stock *from* to fulfill orders.
```
